clients/src/main/java/org/apache/kafka/common/telemetry/internals/LastValueTracker.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.telemetry.internals;
+
+import java.time.Instant;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * A LastValueTracker uses a ConcurrentMap to maintain historic values for a given key, and return
```

Comment:
For my own education: Why do we need a `ConcurrentMap`?

Reply:
As @xvrl mentioned in the comment her that the operations has to be thread safe: https://github.com/apache/kafka/pull/14620#discussion_r1375028442

Reply:
@mjsax I think you meant why `ConcurrentMap` in context of why not just `Map` or mention `ConcurrentHashMap`, the reason why `ConcurrentMap` was used because of the method `putIfAbsent` was being used which is not part of `Map` API.
As I have removed `AtomicReference` and moved from `putIfAbsent` to `put` hence I have also moved from `ConcurrentMap` to `Map`, implementation is still same `ConcurrentHashMap`, I have corrected the comment accordingly.